src/widgets/skills.ts

```diff
@@ -1,29 +1,126 @@
 import buildCard from '../components/card'
 import buildGradientBox from '../components/gradient-box'
 import { findData } from '../utils'
-import errorWidget from './error'
 import languageData from '../data/languages'
+import frameworks from '../data/frameworks'
+import libraries from '../data/libraries'
+import tools from '../data/tools'
+import softwareIDEs from '../data/software-ides'
 
+/**
+ * Builds the skill widget page
+ * 
+ * This method builds a SVG file that contains all the languages, frameworks,
+ * libraries, tools and other software that can be found in the data folder and
+ * are passed by the calling method.
+ * 
+ * @param languagesString The string with all the languages
+ * @param frameworksString The string with all the frameworks
+ * @param librariesString The string with all the libraries
+ * @param toolsString The string with all the tools
+ * @param softwareString The string with all the software
+ * @param includeNames The boolean that determines whether or not to include names
+ * @returns The SVG with all the skills that were passed.
+ */
 export default function skillsWidget(
-    skillsString: string,
-    includeNames: boolean
+    languagesString?: string,
+    frameworksString?: string,
+    librariesString?: string,
+    toolsString?: string,
+    softwareString?: string,
+    includeNames?: boolean,
+
 ): string {
-    const skillList: Array<string> = skillsString.split(',')
 
-    if (skillList === undefined) {
-        return errorWidget('Skills', '-24%', 'Languages are undefined!', '-28%')
+    const BASE_HEIGHT = 125
+    const BASE_WIDTH = 812
+    const FIRST_ROW = 90 
+    const ROW = 114
+    const PAD = 60
+    
+    if (languagesString === undefined || languagesString === null) {
+        languagesString = 'undefined'
+    }
+    if (frameworksString === undefined || frameworksString === null) {
```

Comment:
Btw, why are we even setting the strings to `'undefined'` here? They are already null/undefined so can just be checked as a boolean statement right?

Reply:
True but since they are optional parameters they need to be converted into a string such that the split function can be called on it